### ISSUE TITLE
"Alt text" within the Cover Block is not updating

### ISSUE THREAD

**Issue by jromales** [WordPress/gutenberg#35095]
### Description

What's currently happening: 
* If I create a new post/page, add a Cover block, and insert an image from the Media Library, the existing `Alt text` is copied over; It will show the same info in the `Alt text` in Cover block. When I update the `Alt text` within Media Library, the new `Alt text` isn't copied over to `Alt text` field in Cover block; It will still show the old one. 

What I expect to happen: 
* The `Alt text` should be synced whenever it's updated whether through the Media Library or through the Cover block settings. 

### Step-by-step reproduction instructions

1. Enable Gutenberg plugin.
2. Upload an image in the Media Library,  edit the image, and add something in the `Alt text` field. 
3. Create a new post or page and add a Cover block. 
4. Within that Cover block, insert the image from the Media Library. 
5. Hover and select the Cover block from the post/page to see the `Alt text` field in the block settings. 
6. Change the `Alt text` within Media Library or vice versa. 

### Screenshots, screen recording, code snippet

![Screen Shot 2021-09-24 at 1 01 05 AM](https://user-images.githubusercontent.com/89536719/134551527-eb90500b-3768-4865-a896-2861c602df59.png)

![Screen Shot 2021-09-24 at 1 02 33 AM](https://user-images.githubusercontent.com/89536719/134551720-817fe7f9-9d19-48c2-b635-29fff2d6989b.png)

![Screen Shot 2021-09-24 at 1 03 43 AM](https://user-images.githubusercontent.com/89536719/134551912-76f8b91a-2607-4b9f-ad86-daafe35960a4.png)

![Screen Shot 2021-09-24 at 1 05 50 AM](https://user-images.githubusercontent.com/89536719/134552230-dcb9e533-9635-4983-a102-41c6bb181567.png)

### Environment info

* WordPress version: 5.8.1
* Gutenberg version: 11.5.1
* Browser: Google Chrome
* Macbook Pro with MacOS Big Sur version 11.5.2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by glendaviesnz** [comment]
Thanks for taking time to outline this issue so clearly. This is actually intended behaviour [as explained here](https://github.com/WordPress/gutenberg/issues/28654#issuecomment-772533751). 

Because of the reasons outlined in that past comment, and the additional background discussion [here](https://github.com/WordPress/gutenberg/issues/8941#issuecomment-659664017) and [here](https://core.trac.wordpress.org/ticket/47456) from the accessibility team,  I don't think this decision will be revisited, so closing this issue for now, but feel free to reopen if you want to discuss further.
